_posts/2019-05-20-mockiato-announcement.md

```diff
@@ -0,0 +1,175 @@
+
+```rust
+#[cfg(test)]
+use mockiato::mockable;
+
+#[cfg_attr(test, mockable)]
+trait Greeter {
+    fn greet(&self, name: &str);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_the_world() {
+        let mut greeter = GreeterMock::new();
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .times(1..2);
+        
+        greeter.greet("world");
+    }
+}
+```
+
+## Specifying return values
+The `MethodCallBuilder` also provides a `returns` method.
+
+This `returns` method allow you to specify a return value, which is returned when the corresponding mock method is invoked.
+
+```rust
+#[cfg(test)]
+use mockiato::mockable;
+
+#[cfg_attr(test, mockable)]
+trait Greeter {
+    fn greet(&self, name: &str) -> String;
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_the_world() {
+        let mut greeter = GreeterMock::new();
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .returns(String::from("Hello world"));
+
+        assert_eq!("Hello world", greeter.greet("world"));
+    }
+}
+```
+
+### Specifying different return values for the same call
+For each method of the `Greeter` trait a corresponding `expect_<method_name>_calls_in_order` method is generated.
+
+Ordered expectations can be used together with [`times`](#expecting-a-method-call-a-certain-amount-of-times).
+
+```rust
+#[cfg(test)]
+use mockiato::mockable;
+
+#[cfg_attr(test, mockable)]
+trait Greeter {
+    fn greet(&self, name: &str) -> String;
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_the_world() {
+        let mut greeter = GreeterMock::new();
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .returns(String::from("Hello world"));
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .returns(String::from("Hi world"));
+
+        greeter.expect_greet_calls_in_order();
+
+        assert_eq!("Hello world", greeter.greet("world"));
+        assert_eq!("Hi world", greeter.greet("world"));
+    }
+}
+```
+
+## Limitations
+
+| Name | Description | Tracking Issue |
+| ---- | ----------- | -------------- |
+| `Sync` / `Send` | Mocks are neither sync, nor send. | [#106](https://github.com/myelin-ai/mockiato/issues/106) |
+| Lifetimes on traits | Lifetimes are supported in the method signature, but not on the trait itself. | [#117](https://github.com/myelin-ai/mockiato/issues/117) |
+| Reference to generics | References to types containing a generic type are currently not supported due to a limitation in codegen involving lifetimes. | [#123](https://github.com/myelin-ai/mockiato/issues/123) |
+| Stable support | Mockiato requires nightly because we use the unstable `proc_macro_diagnostics` API to print helpful messages. Printing expected calls requires the unstable feature `specialization`. | — |
```

Comment:
```suggestion
| Stable support | Mockiato requires nightly because we use the unstable `proc_macro_diagnostics` API to print helpful messages. Printing expected calls requires the unstable feature `specialization`. | [#158](https://github.com/myelin-ai/mockiato/pull/158) |
```